include/fmt/format.h

```diff
@@ -778,6 +778,32 @@ FMT_INLINE auto make_args_checked(const S& fmt,
   return {args...};
 }
 
+// compile-time support
+namespace detail_exported {
+#if FMT_USE_NONTYPE_TEMPLATE_PARAMETERS
+template <typename Char, size_t N> struct fixed_string {
+  constexpr fixed_string(const Char (&str)[N]) {
+    detail::copy_str<Char, const Char*, Char*>(static_cast<const Char*>(str),
+                                               str + N, data);
+  }
+  Char data[N]{};
+};
+#endif
+
+// Converts a compile-time string to basic_string_view.
+template <typename Char, size_t N>
+constexpr auto string_to_view(const Char (&s)[N]) -> basic_string_view<Char> {
+  // Remove trailing NUL character if needed. Won't be present if this is used
+  // with a raw character array (i.e. not defined as a string).
+  return {s, N - (std::char_traits<Char>::to_int_type(s[N - 1]) == 0 ? 1 : 0)};
+}
+template <typename Char>
+constexpr auto string_to_view(detail::std_string_view<Char> s)
+    -> basic_string_view<Char> {
+  return {s.data(), s.size()};
+}
```

Comment:
These are only used in macros that are not exported so probably should be in the `detail` namespace, not `detail_exported`. Also please keep the old names (`compile_string_to_view`).

Reply:
This is not correct. These overloads _must_ be visible or `xchar.h`:231 will fail to compile. And there is _no_ workaround for that.
I will change the name. Even though nothing is compiled here but just a view to the stored data created.